tests/test_git_utils.py

```diff
@@ -19,6 +19,9 @@ def _mocked_scm() -> MagicMock:
 
 def test_git_push_tag(mocked_scm):
     tag_name = "test_tag"
+    mocked_scm.push_refspecs = MagicMock(
+        return_value={f"refs/tag/{tag_name}": SyncStatus.SUCCESS}
+    )
```

Comment:
GTO test suite mocks out a lot of git calls, it would be better to actually set up git remote test fixtures (like we have in scmrep and dvc) and ensure that GTO is actually pushing/fetching as expected